packages/react/src/components/table/table.tsx

```diff
@@ -72,6 +73,34 @@ function getHeading<T extends object>(column: HeaderGroup<T>, stickyHeader: bool
     );
 }
 
+const StyledFooter = styled.td<{ sticky: boolean }>`
+    ${({ sticky }) => sticky && css`
+        background-color: ${({ theme }) => theme.greys.white};
+        position: sticky;
+    `}
+`;
+
+function getFooter<T extends object>(column: HeaderGroup<T>, stickyFooter: boolean): ReactElement | null {
+    if (column.Footer === '') {
```

Comment:
`Footer: ''` really communicates an intent of having an empty (but existing) footer column. I would not use that to remove a portion of a previously declared footer.

From a user standpoint, it would be clearer to define a multi-column footer in the first column, then omit the property completely in subsequent columns. It would be very close to what they would do if they were using standard html:
```html
<tr>
    <td>a</td>
    <td>b</td>
</tr>
<tr>
    <td colspan="2">Total: xx</td>
    <!-- Nothing here -->
</tr>
``` 
Would the solution with `Footer.name === 'emptyRenderer'` work in that case or was there other concerns?

Reply:
I was thinking about another solution, so what we all want are:

- Still render the TD if user didn't declare a Footer (this is the default behavior of react-table) and what Carla don't want to lose (I tend to agree now). With `Footer.name === 'emptyRenderer'` you are forced to have `Footer: ''` everywhere that are "meaningless" to the user.
- We don't want `Footer: ''` to remove a cell, because it communicate the intent of having an existing empty cell.

I think this solution would fix both (the user must set the colSpan on the first column of the span but it's a good rule).

```
if (colSpan > 1) {
  colSpan--;
  return null;
}

if (column.footerColSpan) {
  colSpan = column.footerColSpan
}
```

So with that

```
{
    Header: 'Column 1',
    accessor: 'column1',
},
{
    Header: 'Column 2',
    accessor: 'column2',
},
{
    Header: 'Column 3',
    accessor: 'column3',
    Footer: 'Footer 3',
},

<td></td>
<td></td>
<td>Footer 3</td>
```

```
{
    Header: 'Column 1',
    accessor: 'column1',
},
{
    Header: 'Column 2',
    accessor: 'column2',
    Footer: "Hello",
    footerColSpan: 2,
},
{
    Header: 'Column 3',
    accessor: 'column3',
},

<td></td>
<td colSpan="2">Hello</td>
```

Reply:
I still think we need to have some way to tell which cell will be removed to adapt the colspan.
Another suggestion would be: cells with footerColspan = 0 means remove this specific cell.
If we don't define it, it will be difficult to decide which cell to remove, the one before the cell with colSpan bigger than 1 or after?

```
{
    Header: 'Column 1',
    accessor: 'column1',
    Footer: <Button label="Button" buttonType="primary" />,
    footerColSpan: 2,

},
{
    Header: 'Column 2',
    accessor: 'column2',
    footerColSpan: 0,
},
{
    Header: 'Numbers',
    accessor: 'numbers',
    Footer: (value) => {
        const total = useMemo(
            () => value.rows.reduce((sum, row) => row.values.numbers + sum, 0),
            [value.rows],
        );

        return `Total: ${total} `;
    },
},
```

This way the cell you want to expand will have a footerColSpan > 1, 
The cell without the footerColSpan automatically will have colSpan = 1 ,
The cell with footerColSpan > 1 will stretch and have the colScpan as set in the footerColSpan prop.

What do you guys think, it will fix the issues, and also we can define which one to remove, which one to stretch and which cell to be left as it is.

Reply:
Hmm, I find that having to put `footerColSpan: 0` would add complexity, it's the thing we wanted to avoid, but I can see some use case that might need it. So overall, I think we can go with that until we could find something better.

I also though about something completely on different path just to bring more idea, but I'm not sure about it.
`<Table footerGrouping={[1, 3, 1]}>`

Reply:
Yes that could work too, just need to implement and test. 
we could talk about it when we come to an improvement. 
There are other implementations coming to tables maybe this will be more clear in the future